src/registry-transformer/index.ts

```diff
@@ -233,18 +233,38 @@ class Visitor {
 				this.needsLoadable = true;
 				this.ctorCountMap.set(text, (this.ctorCountMap.get(text) || 0) - 1);
 				if (ts.isJsxElement(inputNode)) {
-					const openingElement = ts.updateJsxOpeningElement(
-						node as ts.JsxOpeningElement,
-						ts.createIdentifier(fakeComponentName),
-						attrs
-					);
+					let openingElement;
+
+					if (ts.updateJsxOpeningElement.length === 3) {
+						openingElement = ts.updateJsxOpeningElement(
+							node as ts.JsxOpeningElement,
+							ts.createIdentifier(fakeComponentName),
+							attrs
+						);
+					} else {
+						openingElement = (ts as any).updateJsxOpeningElement(
+							node as ts.JsxOpeningElement,
+							ts.createIdentifier(fakeComponentName),
+							undefined,
```

Comment:
To properly support property type checking this should be `node.typeArguments`. Passing undefined is erase the type arguments if provided.

```typescript
    export function updateJsxOpeningElement(node: JsxOpeningElement, tagName: JsxTagNameExpression, typeArguments: ReadonlyArray<TypeNode> | undefined, attributes: JsxAttributes) {
        return node.tagName !== tagName
            || node.typeArguments !== typeArguments
            || node.attributes !== attributes
            ? updateNode(createJsxOpeningElement(tagName, typeArguments, attributes), node)
            : node;
    }
```